frame/support/src/traits/hooks.rs

```diff
@@ -38,7 +38,9 @@ pub trait OnInitialize<BlockNumber> {
 	}
 }
 
-#[impl_for_tuples(30)]
+#[impl_for_tuples(64)]
+#[cfg_attr(feature = "tuples-96", impl_for_tuples(96))]
+#[cfg_attr(feature = "tuples-128", impl_for_tuples(128))]
```

Comment:
this will do the impl tuples multiple times?

For example both 64 and 128? So 64 also needs a feature flag?

Reply:
Good point, didn't try this, but possible yes: `default = ["std", "tuples-64"]` might work.

Reply:
I agree with @shawntabrizi here, if you don't feature gate 64, it's gonna be applied along with 96 or 128 when they're enabled.

Reply:
aghh do you have anything better than `#[cfg_attr(all(not(feature = "tuples-96"), not(feature = "tuples-128")), impl_for_tuples(64) )]` in mind?

Reply:
A bit janky, but we end up with: 

```
#[cfg_attr(all(not(feature = "tuples-96"), not(feature = "tuples-128")), impl_for_tuples(64))]
#[cfg_attr(feature = "tuples-96", impl_for_tuples(96))]
#[cfg_attr(feature = "tuples-128", impl_for_tuples(128))]
```

This is necessary though, and indeed only now build with `--features tuple-96` work. 

Even now, if you enabled both `tuples-96` and `tuples-128` you get an error. You can fix that as well, but the `cfg` syntax would be horrible. 

We could argue that the users should not enable both features, but given that this good [explanation about features being additive](https://stackoverflow.com/questions/44561643/how-can-i-make-two-features-enabled-lead-to-a-conflict-in-rust), that would be a wrong assumption on the users.

Reply:
> Even now, if you enabled both `tuples-96` and `tuples-128` you get an error. You can fix that as well, but the `cfg` syntax would be horrible.

Not sure it would be that horrible.

```
#[cfg_attr(all(not(feature = "tuples-96"), not(feature = "tuples-128")), impl_for_tuples(64))]
#[cfg_attr(all(feature = "tuples-96", not(feature = "tuples-128")), impl_for_tuples(96))]
#[cfg_attr(feature = "tuples-128", impl_for_tuples(128))]
```

That should work?

Reply:
Right, not much worse, applying now.

Reply:
https://github.com/paritytech/substrate/pull/11813/commits/9bc95fd78db2089b427e64166c2e90ae4371b7a1